src/fsort.c

```diff
@@ -252,11 +252,13 @@ SEXP fsort(SEXP x, SEXP verboseArg) {
 
     if (verbose) {
       Rprintf(_("Top 20 MSB counts: ")); for(int i=0; i<MIN(MSBsize,20); i++) Rprintf(_("%"PRId64" "), (int64_t)msbCounts[order[i]]); Rprintf(_("\n"));
-      Rprintf(_("Reduced MSBsize from %zu to "), MSBsize);
     }
-    while (MSBsize>0 && msbCounts[order[MSBsize-1]] < 2) MSBsize--;
-    if (verbose) {
-      Rprintf(_("%zu by excluding 0 and 1 counts\n"), MSBsize);
+    {
+      size_t oldMSBsize = MSBsize;
+      while (MSBsize>0 && msbCounts[order[MSBsize-1]] < 2) MSBsize--;
+      if (verbose) {
+        Rprintf(_("Reduced MSBsize from %zu to %zu by excluding 0 and 1 counts\n"), oldMSBsize, MSBsize);
```

Comment:
is it possible `oldMSBsize == MSBsize`? If so, maybe we also branch here on `verbose && oldMSBsize != MSBsize`?

WDYT of using a helper here too, `void shrinkMSB(MSBsize, msbCounts, order, verbose)`, that way we don't need to add an indent level to get the restricted scope of `oldMSBsize` too?

Reply:
Made it into a helper in [014a3ae](https://github.com/Rdatatable/data.table/pull/6504/commits/014a3aea423d9d799e33d697b9030ac53eccf9c4).